### PR TITLE
rename `color_function`

### PR DESCRIPTION
**Pull request by deargle**
rename it to `color_values`; raise deprecation warning if old used.

### CODE REVIEW

**Comment by deargle** [comment]
@sauln a micro step on the way to the other stuff I'm going to do with vis

**Comment by deargle** [comment]
I renamed it everywhere except for in the `deprecated/` folder. I also renamed `_color_function` in visuals to be `_node_color_function`, because I intend to make that customizeable too, beyond the current which is to just take the mean.

File: test/test_visuals.py [hunk 1]
```diff
@@ -162,13 +164,29 @@ def test_format_meta_with_meta(self):
         fmt = format_meta(graph, cm)
         assert fmt["custom_meta"] == cm
 
+    def test_color_function_deprecated_replaced(self):
```
Comment (deargle) [inline test/test_visuals.py:167]: @sauln testing that the deprecation warning is raised

File: kmapper/visuals.py [hunk 2]
```diff
@@ -487,8 +487,8 @@ def _format_tooltip(
     return tooltip
 
 
-def _color_function(member_ids, color_function):
-    return np.mean(color_function[member_ids])
+def _node_color_function(member_ids, color_values):
```
Comment (deargle) [inline kmapper/visuals.py:490]: renamed this function

File: kmapper/kmapper.py [hunk 3]
```diff
@@ -601,17 +602,18 @@ def _remove_duplicate_nodes(self, nodes):
...
     def _summary(self, graph, time):
-        # TODO: this summary is dependant on the type of Nerve being built.
+        # TODO: this summary is dependent on the type of Nerve being built.
         links = graph["links"]
         nodes = graph["nodes"]
         nr_links = sum(len(v) for k, v in links.items())
 
         print("\nCreated %s edges and %s nodes in %s." % (nr_links, len(nodes), time))
 
+    @deprecated_alias(color_function='color_values')
```
Comment (deargle) [inline kmapper/kmapper.py:612]: deprecation decorator

File: kmapper/drawing.py [hunk 4]
```diff
@@ -41,6 +41,13 @@ def draw_matplotlib(g, ax=None, fig=None, layout="kk"):
 
     """
     import networkx as nx
+    import os
+    # https://stackoverflow.com/a/50089385/5917194
+    import matplotlib as mpl
+    if os.environ.get('DISPLAY','') == '':
```
Comment (deargle) [inline kmapper/drawing.py:47]: this is necessary to get tox to run the tests. `.travis.yml` has something similar for handingly $DISPLAY

File: examples/output/cat.html [hunk 5]
```diff
@@ -396,9 +396,9 @@ <h3>n_cubes </h3> <p> 15</p>
     
         <h3>perc_overlap </h3> <p> 0.2</p>
     
-        <h3>clusterer </h3> <p> DBSCAN(algorithm='auto', eps=0.1, leaf_size=30, metric='euclidean',<br>    metric_params=None, min_samples=5, n_jobs=None, p=None)</p>
+        <h3>clusterer </h3> <p> DBSCAN(eps=0.1)</p>
```
Comment (deargle) [inline examples/output/cat.html:399]: I'm not sure why these got overwritten. I can revert all changes to `examples`. Must have happened when I was building the docs locally.

Comment (deargle) [reply examples/output/cat.html:399]: Actually, this would have something to do with the template code that builds the examples. Hmm.

Comment (sauln) [reply examples/output/cat.html:399]: I think that sklearn changed the way `__repr__` works and only shows the values that aren't the default.  Not a worry.

File: kmapper/kmapper.py (hunk 3 above)
Comment (sauln) [reply kmapper/kmapper.py:612]: ooh I like that 👍

File: kmapper/plotlyviz.py [hunk 6]
```diff
@@ -255,7 +255,7 @@ def scomplex_to_graph(
 
 def get_mapper_graph(
```
Comment (sauln) [inline kmapper/plotlyviz.py:256]: Does this need the deprecated decorator as well?

**Comment by deargle** [comment]
Okay, I deprecated all of the non-`_` ones also, and added a few more tests.

**Comment by sauln** [comment]
Looks great!

**Pull request merged**